File: framework/java/src/main/java/soda/unittest/work/parse/CharListConverter.java
package soda.unittest.work.parse;

import java.util.List;
import java.util.stream.Collectors;

public class CharListConverter implements ObjectConverter<List<Character>, List<String>> {
    @Override
    public List<Character> fromJsonSerializable(List<String> strs) {
        return strs.stream().map(s -> s.charAt(0)).collect(Collectors.toList());
    }

    @Override
    public List<String> toJsonSerializable(List<Character> chs) {
        return chs.stream().map(c -> "" + c).collect(Collectors.toList());
    }
}

File: framework/scala/src/main/scala/soda/scala/unittest/CharListConverter.scala
package soda.scala.unittest
import play.api.libs.json.{JsValue, Json}

class CharListConverter extends ObjectConverter[List[Char]] {

  override def fromJsonSerializable(js: JsValue): List[Char] = {
    CharListConverter.s2c(js.as[List[String]])
  }

  override def toJsonSerializable(chars: List[Char]): JsValue = {
    Json.toJson(CharListConverter.c2s(chars))
  }

}

object CharListConverter {
  def s2c(s: List[String]): List[Char] = s.map(_.charAt(0))
  def c2s(c: List[Char]): List[String] = c.map("" + _)
}
